### ISSUE TITLE
The use of instance_contrastive_loss

### ISSUE THREAD

**Issue by chenxiaodanhit** [salesforce/CoST#5]
Thanks for your great work! 
Could you please explain the instance contrastive loss as written in cost.py?

> logits = torch.tril(sim, diagonal=-1)[:, :, :-1]  # T x 2B x (2B-1)
logits += torch.triu(sim, diagonal=1)[:, :, 1:]
logits = -F.log_softmax(logits, dim=-1)
i = torch.arange(B, device=z1.device)
loss = (logits[:, i, B + i - 1].mean() + logits[:, B + i, i].mean()) / 2
 
The above defination is different from the L_{amp} and L_{phase} noted in the paper.

Look forward to your reply! Thanks in advance!

**Comment by gorold** [comment]
Hey @chenxiaodanhit, thanks for your interest in our work. The above implementation is equivalent to L_{amp} and L_{phase} as defined in the paper. To explain, the `torch.tril` and `torch.triu` lines ensure that the similarity matrix when `i == j` is omitted, and `logits` calculates the log softmax term for all indices, but we have to perform selection on the appropriate indices, as performed by the last line of code. Hope this helps.

**Comment by chenxiaodanhit** [comment]
Thanks for your reply!
